### ISSUE TITLE
Game crash shulker entity 1.18.1

### ISSUE THREAD

**Issue by WachumaFox** [Sollace/Presence-Footsteps#135]
https://crashy.net/TeJPEMr3XW1skJqW0D9M 
I got a really odd crash while going near an end city
if it helps, I was flying near it with Icarus modded wings

I am unable to join the singleplayer world without the crash

Minecraft 1.18.1 
Fabric 0.13.3
Fabric API 0.46.6+1.18
PresenceFootsteps-1.4.0

**Comment by Sollace** [comment]
Can you please upload the crash log over here:

https://pastebin.com

Also current version of PresenceFootsteps is 1.5.0 for Minecraft 1.18.2

**Comment by WachumaFox** [comment]
https://pastebin.com/EN3gKzLG

**Comment by Sollace** [comment]
This looks like it's a conflicting mod. What mods do you have that mess with boats and/or shulkers?

```
java.lang.NullPointerException: Cannot invoke "net.minecraft.class_1297.method_5864()" because "entity" is null
	at Not Enough Crashes deobfuscated stack trace.(1.18.1+build.22)
	at net.minecraft.entity.mob.ShulkerEntity.redirect$hci000$fixTerraformBoatHeightOffset(ShulkerEntity:2377)
	at net.minecraft.entity.mob.ShulkerEntity.getHeightOffset(ShulkerEntity:335)
	at ```

**Comment by spoorn** [comment]
Same thing happens for me using 1.5.0 on Minecraft 1.18.2 using my modpack: https://www.curseforge.com/minecraft/modpacks/spoornpack-fabric/files/3752322

**Comment by spoorn** [comment]
I added a null check to prevent this from happening in one of my mods: https://www.curseforge.com/minecraft/mc-mods/quality-of-life-qol-fixes/files/3755351

The getHeightOffset method isn't NPE safe
